src/transaction/confirm.rs

```diff
@@ -0,0 +1,601 @@
+//! Transaction confirmation implementation. This is a re-implementation of
+//! `web3` confirmation future to fix issues with development nodes like Ganache
+//! where the transaction gets mined right away, so waiting for 1 confirmation
+//! would require another transaction to be sent so a new block could mine.
+//! Additionally, waiting for 0 confirmations in `web3` means that the tx is
+//! just sent to the mem-pool but does not wait for it to get mined. Hopefully
+//! some of this can move upstream into the `web3` crate.
+
+use crate::errors::ExecutionError;
+use crate::future::{CompatCallFuture, MaybeReady, Web3Unpin};
+use futures::compat::{Compat01As03, Future01CompatExt};
+use futures::future::{self, TryJoin};
+use futures::ready;
+use futures_timer::Delay;
+use std::fmt::{self, Debug, Formatter};
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::time::Duration;
+use web3::api::{CreateFilter, FilterStream, Web3};
+use web3::futures::stream::{Skip as Skip01, StreamFuture as StreamFuture01};
+use web3::futures::Stream as Stream01;
+use web3::types::{TransactionReceipt, H256, U256};
+use web3::Transport;
+
+/// A struct with the confirmation parameters.
+#[derive(Clone, Debug)]
+pub struct ConfirmParams {
+    /// The number of blocks to confirm the transaction with. This is the number
+    /// of blocks mined on top of the block where the transaction was mined.
+    /// This means that, for example, to just wait for the transaction to be
+    /// mined, then the number of confirmations should be 0. Positive non-zero
+    /// values indicate that extra blocks should be waited for on top of the
+    /// block where the transaction was mined.
+    pub confirmations: usize,
+    /// The polling interval. This is used as the interval between consecutive
+    /// `eth_getFilterChanges` calls to get filter updates, or the interval to
+    /// wait between confirmation checks in case filters are not supported by
+    /// the node (for example when using Infura over HTTP(S)).
+    pub poll_interval: Duration,
+    /// The maximum number of blocks to wait for a transaction to get confirmed.
+    pub block_timeout: usize,
```

Comment:
Does it make sense for this to be an option? Given that the tx will be in the mem pool you might not want to stop waiting for it (I realize you could use a very high value here).

Reply:
We can do that, it’s not too hard.